backend/entities/application_entity.py

```diff
@@ -0,0 +1,386 @@
+
+    # Name for the applications table in the PostgreSQL database
+    __tablename__ = "application"
+
+    # Unique ID for the application
+    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
+
+    # The user associated with the application
+    # NOTE: This field establishes a one-to-many relationship between the user and application tables.
+    user_id: Mapped[int] = mapped_column(ForeignKey("user.id"), nullable=True)
+    user: Mapped["UserEntity"] = relationship(back_populates="applications")
+
+    # Set up for single-table inheritance (assign unique polymorphic identity)
+    type = Column(String(50))
+    __mapper_args__ = {"polymorphic_identity": "application", "polymorphic_on": type}
+
+    @classmethod
+    def from_model(cls, model: Application) -> Self:
+        """
+        Class method that converts an `Application` model into a `ApplicationEntity`
+
+        Parameters:
+            - model (Application): Model to convert into an entity
+        Returns:
+            ApplicationEntity: Entity created from model
+        """
+        return cls(
+            id=model.id,
+            user_id=model.user_id,
+        )
+
+    def to_model(self) -> Application:
+        """
+        Converts an `ApplicationEntity` object into an `Application` model object
+
+        Returns:
+            Application: `Application` object from the entity
+        """
+        return Application(
+            id=self.id,
+            user_id=self.user_id,
+        )
+
+    def to_details_model(self) -> ApplicationDetails:
+        """
+        Converts a `ApplicationEntity` object into a `ApplicationDetails` model object
+
+        Returns:
+            ApplicationDetails: `ApplicationDetails` object from the entity
+        """
+        return ApplicationDetails(
+            id=self.id, user_id=self.user_id, user=self.user.to_model()
+        )
+
+
+class UTAEntity(ApplicationEntity):
+    """Serves as the database model schema for applications specific to Undergraduate TA's"""
+
+    # Application properties (columns in the database table) specific to UTA Applications
+
+    # Academic Hours student plans to take
+    academic_hours: Mapped[int] = mapped_column(Integer, nullable=True)
+
+    # Extracurriculars student is a part of
+    extracurriculars: Mapped[str] = mapped_column(String, nullable=True)
+
+    # Expected graduation
+    expected_graduation: Mapped[str] = mapped_column(String, nullable=True)
+
+    # Program pursued
+    program_pursued: Mapped[str] = mapped_column(String, nullable=True)
+
+    # Other programs being pursued
+    other_programs: Mapped[str] = mapped_column(String, nullable=True)
+
+    # GPA
+    gpa: Mapped[float] = mapped_column(Float, nullable=True)
+
+    # COMP GPA
+    comp_gpa: Mapped[float] = mapped_column(Float, nullable=True)
+
+    # Do they want to do this as COMP 227?
+    comp_227: Mapped[str] = mapped_column(String, nullable=True)
```

Comment:
This would not be a bool because the options vary on the semester. I.e:
1. Not taking COMP 227
2. Taking COMP 227 (no compensation)
3. Doing a split
With the options potentially changing in the future, we decided to store this as a string. Do you think we should change it to something like an ENUM?

Reply:
@alshayefaziz This might be better off as an `Enum`! A good example for this is `RosterRole` in the academics feature